exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -387,3 +411,24 @@ func (r *ClusterResourceSetReconciler) clusterToClusterResourceSet(o handler.Map
 	}
 	return result
 }
+
+// resourceToClusterResourceSet is mapper function that maps resources to ClusterResourceSet
+func (r *ClusterResourceSetReconciler) resourceToClusterResourceSet(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+	acc, err := meta.Accessor(o.Object)
+	if err != nil {
+		return nil
+	}
+
+	crsList := &addonsv1.ClusterResourceSetList{}
+	if err := r.Client.List(context.Background(), crsList, client.InNamespace(acc.GetNamespace())); err != nil {
+		return nil
+	}
+
+	for _, crs := range crsList.Items {
+		name := client.ObjectKey{Namespace: acc.GetNamespace(), Name: crs.Name}
+		result = append(result, ctrl.Request{NamespacedName: name})
```

Comment:
Shouldn't this check that the `acc` is in the list of Resources for the given ClsuterResourceSet? Currently this looks to return all ClusterResourceSets in the same namespace as the ConfigMap/Secret that is triggering the reconciliation.

Reply:
Yes, it would be better to do that check here.